src/NuGet.Clients/VisualStudio.Facade/NuGetUIThreadHelper.cs

```diff
@@ -33,28 +33,29 @@ public static JoinableTaskFactory JoinableTaskFactory
         /// Retrieve the CPS enabled JoinableTaskFactory for the current version of Visual Studio.
         /// This overrides the default VsTaskLibraryHelper.ServiceInstance JTF.
         /// </summary>
-        public static void SetJoinableTaskFactoryFromService(IComponentModel componentModel)
+        public static void SetJoinableTaskFactoryFromService(IProjectServiceAccessor projectServiceAccessor)
         {
-            if (componentModel == null)
+            if (projectServiceAccessor == null)
             {
-                throw new ArgumentNullException(nameof(componentModel));
+                throw new ArgumentNullException(nameof(projectServiceAccessor));
             }
 
-            _joinableTaskFactory = new Lazy<JoinableTaskFactory>(() =>
+            if (_joinableTaskFactory == null)
             {
-                var projectServiceAccessor = componentModel.GetService<IProjectServiceAccessor>();
-
+                _joinableTaskFactory = new Lazy<JoinableTaskFactory>(() =>
```

Comment:
This is still a race condition. You are not guaranteed to execute the null check and creating the lazy without being interrupted. You need to use a `lock` region.